code/game/objects/items/tools/maintenance_tools.dm

```diff
@@ -457,9 +457,12 @@
 	///Are we currently recharging something.
 	var/recharging = FALSE
 
-/obj/item/tool/handheld_charger/Initialize(mapload)
+/obj/item/tool/handheld_charger/update_icon_state()
 	. = ..()
-	cell = null
+	if(cell)
+		icon_state = "handheldcharger_black"
```

Comment:
You don't need to call parent for this.

Also its fine for in this case, but keep in mind that specifying exactly what icon states to use can break stuff down the line.
for example we have a different icon set for hand chargers (unused), but if they were used this would invalidate them.

What its better is just to call initial(icon_state) or "[initial(icon_state)]_empty" for example.